broken on macOS

looks like the switch to c++11 broke something in `exceptions.h` on macOS clang:
```
$ make
[  0%] Built target expand_instantiations_exe
[  0%] Built target doxygen_headers
[  0%] Built target obj_opencascade_inst
[  0%] Building CXX object source/opencascade/CMakeFiles/obj_opencascade_debug.dir/utilities.cc.o
In file included from /Users/davydden/libs-sources/deal.ii/davydden/source/opencascade/utilities.cc:16:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/opencascade/utilities.h:20:
In file included from /Users/davydden/libs-sources/deal.ii/builds/spack/include/deal.II/base/config.h:333:
/Users/davydden/spack/_dealii_suite/include/mpi.h:328:9: warning: 'long long' is a C++11 extension [-Wc++11-long-long]
typedef OMPI_MPI_OFFSET_TYPE MPI_Offset;
        ^
/Users/davydden/spack/_dealii_suite/include/mpi.h:150:30: note: expanded from macro 'OMPI_MPI_OFFSET_TYPE'
#define OMPI_MPI_OFFSET_TYPE long long
                             ^
/Users/davydden/spack/_dealii_suite/include/mpi.h:329:9: warning: 'long long' is a C++11 extension [-Wc++11-long-long]
typedef OMPI_MPI_COUNT_TYPE MPI_Count;
        ^
/Users/davydden/spack/_dealii_suite/include/mpi.h:159:29: note: expanded from macro 'OMPI_MPI_COUNT_TYPE'
#define OMPI_MPI_COUNT_TYPE long long
                            ^
In file included from /Users/davydden/libs-sources/deal.ii/davydden/source/opencascade/utilities.cc:16:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/opencascade/utilities.h:20:
In file included from /Users/davydden/libs-sources/deal.ii/builds/spack/include/deal.II/base/config.h:336:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/numbers.h:21:
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/types.h:47:20: warning: 'long long' is a C++11 extension [-Wc++11-long-long]
  typedef unsigned long long int global_vertex_index;
                   ^
In file included from /Users/davydden/libs-sources/deal.ii/davydden/source/opencascade/utilities.cc:16:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/opencascade/utilities.h:24:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/grid/tria.h:21:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/point.h:21:
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:58:11: error: exception specification of overriding function is more lax than base version
  virtual ~ExceptionBase () noexcept;
          ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/exception:95:13: note: overridden virtual function is here
    virtual ~exception() _NOEXCEPT;
            ^
In file included from /Users/davydden/libs-sources/deal.ii/davydden/source/opencascade/utilities.cc:16:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/opencascade/utilities.h:24:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/grid/tria.h:21:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/point.h:21:
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:58:28: error: expected ';' at end of declaration list
  virtual ~ExceptionBase () noexcept;
                           ^
                           ;
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:75:23: error: exception specification of overriding function is more lax than base version
  virtual const char *what() const noexcept;
                      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/exception:96:25: note: overridden virtual function is here
    virtual const char* what() const _NOEXCEPT;
                        ^
In file included from /Users/davydden/libs-sources/deal.ii/davydden/source/opencascade/utilities.cc:16:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/opencascade/utilities.h:24:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/grid/tria.h:21:
In file included from /Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/point.h:21:
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:75:35: error: expected ';' at end of declaration list
  virtual const char *what() const noexcept;
                                  ^
                                  ;
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:641:3: error: expected ';' at end of declaration list
  DeclExceptionMsg (ExcDivideByZero,
  ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:413:27: note: expanded from macro 'DeclExceptionMsg'
    virtual ~Exception () noexcept {}                                     \
                          ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:641:3: error: member initializer 'arg' does not name a non-static data member or base class
  DeclExceptionMsg (ExcDivideByZero,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:412:56: note: expanded from macro 'DeclExceptionMsg'
    Exception (const std::string &msg = defaulttext) : arg (msg) {}       \
                                                       ^~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:655:3: error: expected ';' at end of declaration list
  DeclException1 (ExcNumberNotFinite,
  ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:431:28: note: expanded from macro 'DeclException1'
    virtual ~Exception1 () noexcept {}                                    \
                           ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:655:3: error: member initializer 'arg1' does not name a non-static data member or base class
  DeclException1 (ExcNumberNotFinite,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:430:35: note: expanded from macro 'DeclException1'
    Exception1 (const type1 a1) : arg1 (a1) {}                            \
                                  ^~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:689:3: error: expected ';' at end of declaration list
  DeclExceptionMsg (ExcOutOfMemory,
  ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:413:27: note: expanded from macro 'DeclExceptionMsg'
    virtual ~Exception () noexcept {}                                     \
                          ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:689:3: error: member initializer 'arg' does not name a non-static data member or base class
  DeclExceptionMsg (ExcOutOfMemory,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:412:56: note: expanded from macro 'DeclExceptionMsg'
    Exception (const std::string &msg = defaulttext) : arg (msg) {}       \
                                                       ^~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:702:3: error: expected ';' at end of declaration list
  DeclException1 (ExcMemoryLeak, int,
  ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:431:28: note: expanded from macro 'DeclException1'
    virtual ~Exception1 () noexcept {}                                    \
                           ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:702:3: error: member initializer 'arg1' does not name a non-static data member or base class
  DeclException1 (ExcMemoryLeak, int,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:430:35: note: expanded from macro 'DeclException1'
    Exception1 (const type1 a1) : arg1 (a1) {}                            \
                                  ^~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:709:3: error: expected ';' at end of declaration list
  DeclExceptionMsg (ExcIO,
  ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:413:27: note: expanded from macro 'DeclExceptionMsg'
    virtual ~Exception () noexcept {}                                     \
                          ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:709:3: error: member initializer 'arg' does not name a non-static data member or base class
  DeclExceptionMsg (ExcIO,
  ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:412:56: note: expanded from macro 'DeclExceptionMsg'
    Exception (const std::string &msg = defaulttext) : arg (msg) {}       \
                                                       ^~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:738:3: error: expected ';' at end of declaration list
  DeclException1 (ExcFileNotOpen,
  ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:431:28: note: expanded from macro 'DeclException1'
    virtual ~Exception1 () noexcept {}                                    \
                           ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:738:3: error: member initializer 'arg1' does not name a non-static data member or base class
  DeclException1 (ExcFileNotOpen,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:430:35: note: expanded from macro 'DeclException1'
    Exception1 (const type1 a1) : arg1 (a1) {}                            \
                                  ^~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:750:3: error: expected ';' at end of declaration list
  DeclExceptionMsg (ExcNotImplemented,
  ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:413:27: note: expanded from macro 'DeclExceptionMsg'
    virtual ~Exception () noexcept {}                                     \
                          ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:750:3: error: member initializer 'arg' does not name a non-static data member or base class
  DeclExceptionMsg (ExcNotImplemented,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:412:56: note: expanded from macro 'DeclExceptionMsg'
    Exception (const std::string &msg = defaulttext) : arg (msg) {}       \
                                                       ^~~~~~~~~
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:782:3: error: expected ';' at end of declaration list
  DeclExceptionMsg (ExcInternalError,
  ^
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/exceptions.h:413:27: note: expanded from macro 'DeclExceptionMsg'
    virtual ~Exception () noexcept {}                                     \
                          ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
3 warnings and 20 errors generated.
make[2]: *** [source/opencascade/CMakeFiles/obj_opencascade_debug.dir/utilities.cc.o] Error 1
make[1]: *** [source/opencascade/CMakeFiles/obj_opencascade_debug.dir/all] Error 2
make: *** [all] Error 2
```

--- a/cmake/checks/check_01_cxx_features.cmake
+++ b/cmake/checks/check_01_cxx_features.cmake
@@ -325,9 +325,14 @@ RESET_CMAKE_REQUIRED()
 CHECK_CXX_SOURCE_COMPILES(
   "
   #include <memory>
-  int main()
+
+  #if __cplusplus < 201103L
+  #  error \"The compiler does not default to C++11 or newer.\"
+  #endif
+
+  auto main() -> int
   {
-    std::unique_ptr<int> p0;
+    auto p0 = std::unique_ptr<int>();
     auto p1 = std::move(p0);
   }
   "

